<a href="https://colab.research.google.com/github/sergiomar73/nlp-google-colab/blob/main/Embeddings.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
example = 'Oh, so the quantified platform is one of the most advanced communication intelligence in AI powered coaching systems.'

# GPT-3 embeddings

In [1]:
!pip install openai

import os
import openai

openai.organization = "org-EXUSyiiEBGlkutelulU3iC1D"
openai.api_key = "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"

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 43 kB 1.5 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 163 kB 7.6 MB/s 
  Created wheel for openai: filename=openai-0.23.1-py3-none-any.whl size=54527 sha256=0a00f8ed67d007e87b113fc45d18501e361247a0e28a1c8e9f6bbc7bfd6a28a8
  Stored in directory: /root/.cache/pip/wheels/5f/f5/69/860bc773610989840e34f3352d39a4651a76f509cf1610bf22
Successfully built openai


In [4]:
gpt3_embedding = openai.Embedding.create(
    input=example,
    engine="text-similarity-davinci-001")
gpt3_embedding

<OpenAIObject list at 0x7f1ec283def0> JSON: {
  "data": [
    {
      "embedding": [
        -0.003871886758133769,
        0.004901846870779991,
        -0.008754660375416279,
        -0.013198747299611568,
        -0.0008678367012180388,
        0.015830866992473602,
        -0.002692518988624215,
        -0.002630530623719096,
        -0.0015465294709429145,
        -0.014686467126011848,
        -0.00014195726544130594,
        -0.0002743778459262103,
        0.002908683381974697,
        -0.0069935559295117855,
        -0.0015481188893318176,
        0.00044107084977440536,
        -0.014241422526538372,
        -0.005687032360583544,
        -0.004186596721410751,
        0.011831824667751789,
        -0.00150838284753263,
        -0.006898188963532448,
        0.020027000457048416,
        0.009231493808329105,
        0.010420397855341434,
        0.002749738981947303,
        0.02138756588101387,
        0.0006449171341955662,
        -0.008513065055012703,
        -0.00595723

In [5]:
len(gpt3_embedding["data"][0]['embedding'])

12288

Via a REST API endpoint, you can access four types of models from OpenAI:

1.   Ada (1024 dimensions)
2.   Babbage (2048 dimensions)
3.   Curie (4096 dimensions)
4.   Davinci (12288 dimensions)

# BERT embeddings

Dimensions depend on the model

In [ ]:
BERT